src/operations/query/filters/aboveAndBelow.js

```diff
@@ -1,44 +1,25 @@
-/**
- * filters by above and below FHIR search parameters
- * https://www.hl7.org/fhir/search.html#modifiers
- * @param {import('../../common/types').SearchParameterDefinition} propertyObj
- * @param {Object} args
- * @param {string} queryParameter
- * @param {Set} columns
- * @return {Object[]}
- */
-function filterByAboveAndBelow({propertyObj, args, queryParameter, columns}) {
```

Comment:
because these segments are ultimately anded together, we can handle above and below in different passes through the major loop instead of all at once